diktat-rules/src/main/kotlin/com/saveourtool/diktat/ruleset/rules/chapter1/IdentifierNaming.kt

```diff
@@ -189,38 +198,55 @@ class IdentifierNaming(configRules: List<RulesConfig>) : DiktatRule(
                         }
                     }
                 } else if (variableName.text != "_" && !variableName.text.isLowerCamelCase()) {
+                    var isCorrectBackingField = false
+                    if (variableName.text.commonPrefixWith("_").length == 1 && variableName.text.drop(1).isLowerCamelCase()) {
```

Comment:
suggestion to move this new logic to a separate method, something like:
```kotlin
private fun ASTNode.isCorrectBackingField(variableName: ASTNode) {
    val propertyNodes = node.treeParent.getAllChildrenWithType(KtNodeTypes.PROPERTY)
    // bla bla
}
```

Reply:
and please add some comments about your algorithm

Reply:
Done